app/javascript/packs/search.js

```diff
@@ -141,17 +209,26 @@ function includeFilteredCheckedAgenciesInSearch(agencyList) {
   });
 }
 
+// Validates publication date range and creates badge.
+// returns True if successful, false otherswise
 function publicationDateValidation(){
   let startYear = parseInt($("#starting_year").val());
   let endYear = parseInt($("#ending_year").val());
+  if (isNaN(startYear) && isNaN(endYear)) {
+    clearDatesFilter();
+    return true;
```

Comment:
given that there are 4 cases here, I would initialize a variable as either `true/false` and then update the cases to return that variable as that could dry this up a bit.